### ISSUE TITLE
install on macos m1

### ISSUE THREAD

**Issue by glensc** [bulletmark/cdhist#1]
```sh
for _d in /usr/share /usr/local/share; do
    _f="$_d/cdhist/cdhist.rc"
    if [[ -r $_f ]]; then
        echo source $_f
        break
    fi
done
```

this doesn't work as it's installed elsewhere with pip3 from, brew

**Comment by glensc** [comment]
```
$ which pip3
/opt/homebrew/bin/pip3
$ pip3 install cdhist
Collecting cdhist
  Downloading cdhist-2.7-py3-none-any.whl (10 kB)
Installing collected packages: cdhist
Successfully installed cdhist-2.7
$
```

**Comment by glensc** [comment]
```
$ find /opt/homebrew -name cdhist
/opt/homebrew/bin/cdhist
/opt/homebrew/share/cdhist
```

**Comment by bulletmark** [comment]
The SETUP instruction in the README **is very clear** about this. Please note the second sentence in the second paragraph there:

"Note that some installations may place the `cdhist/ `directory elsewhere to the two most common locations listed above so you may need to work out the location from where to source the `cdhist.rc` file."

In the very first sentence of the README it says this is a Linux utility so it should be very apparent that using Homebrew on a Mac is likely to be quite different. It is not feasible for me to add every possible directory which could exist in every possible installation in every possible environment.

Also, that fragment of shell script you have ostensibly posted above from the README will never work as you have inserted a stray `echo` command.

**Comment by glensc** [comment]
@bulletmark the issue was that you update instructions for m1. and the echo is obviously to show what files are loaded. macOS and m1 are pretty common, but I guess you just prefer to drive users away. bye.

**Comment by bulletmark** [comment]
I added commit 2f6092e to improve this (version 2.8). Now the example shell boilerplate should work for all platforms.

**Comment by glensc** [comment]
yeah, 2f6092eda5697451b6710b26b5300af8675f8a53 definitely makes things worse, you should restore the previous solution as fallback option.

```
$ pip3 install cdhist -U # installed 2.9
$ cdhist -s
/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/share/cdhist/cdhist.rc
$ ls -l /opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/share/cdhist/cdhist.rc
ls: cannot access '/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/share/cdhist/cdhist.rc': No such file or directory
```

where it actually is:
```
$  find /opt/homebrew -name cdhist.rc
/opt/homebrew/share/cdhist/cdhist.rc
```

ps: you should make atomic commits and pull requests on separate changes, so could comment on exact thing, not whole massive commit
- https://www.freshconsulting.com/insights/blog/atomic-commits/

**Comment by bulletmark** [comment]
Geez, damn Apple. Why the heck does `sys.prefix` resolve to that.

Anyhow, the current approach certainly works better for most users as it is automatic. Doesn't work for Mac, but it didn't previously either as you argue here. I would just have to add back the comment about having to configure it explicitly for some systems.

**Comment by glensc** [comment]
I have in my .zshrc:

```sh
# cdhist: https://github.com/bulletmark/cdhist
for _d in /usr/share /usr/local/share /opt/homebrew/share; do
    _f="$_d/cdhist/cdhist.rc"
    if [[ -r $_f ]]; then
		source $_f
		unset _f _d
		break
    fi
done
unset _f _d
export FZF_ALT_C_COMMAND="cat $HOME/.cd_history"
```

**Comment by glensc** [comment]
And it's not Damn Apple, but Damn Homebrew:

```
~ ➜ which python3
/opt/homebrew/bin/python3
~ ➜ python3 -c "import sys; print(sys.prefix)"
/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9
~ ➜
```

**Comment by glensc** [comment]
sys.prefix is for python installation path, but pip installation path may be different.

how about you just use `__file__` in `cdhist.py`? `cdhist.rc` is relative to `cdhist.py`:

```
cdhist (master) [★] ✖  find /opt/homebrew/ -name 'cdhist.*'
/opt/homebrew/lib/python3.9/site-packages/__pycache__/cdhist.cpython-39.pyc
/opt/homebrew/lib/python3.9/site-packages/cdhist.py
/opt/homebrew/share/cdhist/cdhist.rc
```

**Comment by bulletmark** [comment]
Where is the `cdhist` executable installed to?

**Comment by bulletmark** [comment]
Just realized you say at the top here that it is installed at `/opt/homebrew/bin/cdhist` so I have made commit 600d0403cb4923441192ad5254486bb52134861c which should work everywhere and have released new version 2.10.

**Comment by glensc** [comment]
yeah, but I had in mind `__file__`, if inserted to `cdhist.py` it will give you the location of cdhist.py. I personally don't trust sys.argv[0], it may be python if invoked differently.

so far your solution seems to work on macOS amd64

```
~ ➜ which cdhist
/usr/local/bin/cdhist
~ ➜ cdhist -s|xargs ls -ld
-rw-r--r-- 1 glen admin 1716 juuli 31 13:49 /usr/local/share/cdhist/cdhist.rc
~ ➜ python3 /usr/local/bin/cdhist -s
/usr/local/share/cdhist/cdhist.rc
~ ➜
```

will check on m1 later
